include/Geometry.h

```diff
@@ -4,16 +4,18 @@
 #include <glm/glm.hpp>
 #include <GLFW/glfw3.h>
 
-struct Geometry
+class Geometry
 {
+public:
 	// OpenGL names for array buffer objects, vertex array object
 	GLuint  vertexBuffer;
 	GLuint  textureBuffer;
 	GLuint  colourBuffer;
 	GLuint  vertexArray;
 
-	std::vector<glm::vec2>& vertices;
-	std::vector<glm::vec3>& colours;
+	std::vector<glm::vec2> vertices;
+	std::vector<glm::vec3> colours;
 
```

Comment:
What's the reason for switching from passing pointers? I'm genuinely curious.

Reply:
Making use of move semantics, we can afford to not store references to the vertex and colour data (not reliable) at the cost of copying the data once during the function call.

Background:
Passing the vertices and colours by value means that we would copy the data twice - once in the function call, once in the assignment.

Initializing geometry with glm::vec references was a first attempt to avoid copying glm::vec data at all. However, this would cause issues as these references would go stale non-deterministically.

C++11 move semantics allows the Geometry to keep a copy of the values (not references) at the cost of copying the data once during the function call. This is worth it since the copies would not be deleted until the Geometry object is destroyed.

Reply:
Alright, that sounds great. Thanks for the explanation.